rct229/rulesets/ashrae9012019/section11/section11rule10.py

```diff
@@ -0,0 +1,457 @@
+                    )
+
+                if (
+                    standby_loss_target_metric_b is not None
+                    and modeled_standby_loss_b is None
+                ):
+                    manual_check_msg.append(
+                        f"Based on the provided details, {standby_loss_target_metric_b} is an expected efficiency metric for this water heater, however it was not provided. Note that the specific requirements of 10 CFR 430 can be found in ASHRAE 90.1 Appendix F Table F-2."
+                    )
+
+                return "\n".join(manual_check_msg)
+
+            def rule_check(self, context, calc_vals=None, data=None):
+                swh_type_b = calc_vals["swh_type_b"]
+                modeled_efficiency_b = calc_vals["modeled_efficiency_b"]
+                modeled_standby_loss_b = calc_vals["modeled_standby_loss_b"]
+                expected_efficiency_b = calc_vals["expected_efficiency_b"]
+                standby_loss_target_b = calc_vals["standby_loss_target_b"]
+                standby_loss_target_metric_b = calc_vals["standby_loss_target_metric_b"]
+
+                standby_loss_complies = (
+                    (standby_loss_target_b is None)
+                    or (
+                        standby_loss_target_metric_b
+                        == SWHEfficiencyMetricOptions.STANDBY_LOSS_FRACTION
+                        and self.precision_comparison["swh_standby_loss_fraction_b"](
+                            modeled_standby_loss_b, standby_loss_target_b
+                        )
+                    )
+                    or (
+                        standby_loss_target_metric_b
+                        == SWHEfficiencyMetricOptions.STANDBY_LOSS_ENERGY
+                        and self.precision_comparison["swh_standby_loss_energy_b"](
+                            modeled_standby_loss_b, standby_loss_target_b
+                        )
+                    )
+                    or (
+                        standby_loss_target_b is not None
+                        and modeled_standby_loss_b < standby_loss_target_b
+                    )
+                )
+
+                return (
+                    swh_type_b in STORAGE_TYPES
+                    and self.precision_comparison["swh_efficiency_b"](
+                        modeled_efficiency_b, expected_efficiency_b
+                    )
+                    and standby_loss_complies
+                )
+
+            def get_fail_msg(self, context, calc_vals=None, data=None):
+                swh_type_b = calc_vals["swh_type_b"]
+
+                if swh_type_b == "OTHER":
+                    return "The water heater type was not recognized, and does not match any of the expected baseline water heater types."
+                else:
+                    return "The modeled efficiency or standby loss for the water heater does not match the expected values."
+
+            def is_tolerance_fail(self, context, calc_vals=None, data=None):
```

Comment:
@weilixu is this method needed?

Reply:
it is probably good to keep one here since we dont know how the precision works in eval function.